### PR TITLE
Grilles can now act as lightning rods for Arcflashes

### PR DESCRIPTION
**Pull request by Skeletonman0**
[Feature] [Qol]

## About the PR 
This PR makes it so grilles can take energy from arcflashes and add a portion of it (affected by matsci stats) to an attached wire.

It also fixes a minor mistake i made with my last PR related to grilles, specifically with the chance of getting shocked by the grilles. This caused the chances to be lower for grilles with no conductivity stat

Currently the only way i have found to actually test this is to hit the grille with buildmode arcflashes, as most things that arcflash only target mobs. I was unsure of if it would be a good idea to change this.

## Why's this needed? 
It could allow for some interesting engineering related gimmicks. especially if more arcflashes become able to hit the grilles.

One thing that could become possible is trapping an arcfiend in a box and powering the station off of that.

## Changelog

```changelog
(u)Skeletonman0
(+)Grilles hit by arcflashes will transfer some of the energy to a connected wire.

### CODE REVIEW

File: code/obj/grille.dm [hunk 1]
```diff
@@ -519,6 +522,18 @@
 
 		return src.electrocute(user, prb, net, ignore_gloves)
 
+	proc/lightningrod(lpower)
+		if (!anchored)
+			return 0
```
Comment (Zonespace27) [inline code/obj/grille.dm:527]:
```suggestion
			return FALSE
```

File: code/obj/grille.dm [hunk 2]
```diff
@@ -519,6 +522,18 @@
...
+		var/net = get_connection()
+		if (!powernets[net])
+			return 0
```
Comment (Zonespace27) [inline code/obj/grille.dm:530]:
```suggestion
			return FALSE
```

File: code/obj/grille.dm [hunk 3]
```diff
@@ -519,6 +522,18 @@
...
+		if (src.material.hasProperty("electrical")) // fun fact: you get -1 if this isnt here
+			powernets[net].newavail += lpower/100*(100-src.material.getProperty("electrical"))
+		else
+			powernets[net].newavail += lpower/100*40 // default to 60
+		return
```
Comment (Zonespace27) [inline code/obj/grille.dm:535]:
```suggestion
```

File: code/obj/grille.dm [hunk 4]
```diff
@@ -519,6 +522,18 @@
...
 		return src.electrocute(user, prb, net, ignore_gloves)
 
+	proc/lightningrod(lpower)
+		if (!anchored)
+			return 0
+		var/net = get_connection()
+		if (!powernets[net])
+			return 0
+		if (src.material.hasProperty("electrical")) // fun fact: you get -1 if this isnt here
+			powernets[net].newavail += lpower/100*(100-src.material.getProperty("electrical"))
```
Comment (Zonespace27) [inline code/obj/grille.dm:532]:
```suggestion
			powernets[net].newavail += lpower/100 * (100 - src.material.getProperty("electrical"))
```

File: code/obj/grille.dm [hunk 5]
```diff
@@ -519,6 +522,18 @@
...
+		else
+			powernets[net].newavail += lpower/100*40 // default to 60
```
Comment (Zonespace27) [inline code/obj/grille.dm:534]:
```suggestion
			powernets[net].newavail += lpower / 100 * 40 // default to 60
```

File: code/obj/grille.dm [hunk 6]
```diff
@@ -277,7 +277,10 @@
 		src.visible_message("<span class='alert'><B>[src] was hit by [AM].</B></span>")
 		playsound(src.loc, 'sound/impact_sounds/Metal_Hit_Light_1.ogg', 100, 1)
 		if (ismob(AM))
-			shock(AM, 100 - (60 - src.material.getProperty("electrical")))  // sure loved people being able to throw corpses into these without any consequences.
+			if(src.material && src.material.hasProperty("electrical"))
```
Comment (ZeWaka) [inline code/obj/grille.dm:280]: ?.

File: code/obj/grille.dm [hunk 7]
```diff
@@ -519,6 +522,17 @@
...
+	proc/lightningrod(lpower)
+		if (!anchored)
+			return FALSE
+		var/net = get_connection()
+		if (!powernets[net])
+			return FALSE
+		if (src.material.hasProperty("electrical")) // fun fact: you get -1 if this isnt here
+			powernets[net].newavail += lpower/100 * (100 - src.material.getProperty("electrical"))
+		else
+			powernets[net].newavail += lpower / 100 * 40 // default to 60
```
Comment (ZeWaka) [inline code/obj/grille.dm:534]: where is 60

File: code/obj/grille.dm (hunk 6 above)
Comment (Skeletonman0) [reply code/obj/grille.dm]: went through and made the stuff i put in use ?.

File: code/obj/grille.dm (hunk 7 above)
Comment (Skeletonman0) [reply code/obj/grille.dm]: changed it to (100 - 60), the "electrical" property itself has default_value set to 60

**Comment by Tarmunora** [comment]
inb4 we accidentally reinvent tesla engine

**Review by ZeWaka — APPROVED** [review]

**Pull request merged**